nautobot/circuits/views.py

```diff
@@ -3,277 +3,282 @@
-    filterset = filters.ProviderFilterSet
+class ProviderViewSet(NautobotViewSet):
+    model = Provider
     table = tables.ProviderTable
-    form = forms.ProviderBulkEditForm
-
-
-class ProviderBulkDeleteView(generic.BulkDeleteView):
-    queryset = Provider.objects.annotate(count_circuits=count_related(Circuit, "provider"))
+    form = forms.ProviderForm
+    filterset_form = forms.ProviderFilterForm
     filterset = filters.ProviderFilterSet
-    table = tables.ProviderTable
-
-
-#
-# ProviderNetwork
-#
-
-
-class ProviderNetworkListView(generic.ObjectListView):
-    queryset = ProviderNetwork.objects.all()
-    filterset = filters.ProviderNetworkFilterSet
-    filterset_form = forms.ProviderNetworkFilterForm
-    table = tables.ProviderNetworkTable
-
-
-class ProviderNetworkView(generic.ObjectView):
-    queryset = ProviderNetwork.objects.all()
+    import_form = forms.ProviderCSVForm
+    bulk_edit_form = forms.ProviderBulkEditForm
+    lookup_field = "slug"
+
+    def get_extra_context(self, request, view_type, instance):
+        if view_type == "detail":
+            circuits = (
+                Circuit.objects.restrict(request.user, "view")
+                .filter(provider=instance)
+                .prefetch_related("type", "tenant", "terminations__site")
+            )
 
-    def get_extra_context(self, request, instance):
-        circuits = (
-            Circuit.objects.restrict(request.user, "view")
-            .filter(Q(termination_a__provider_network=instance.pk) | Q(termination_z__provider_network=instance.pk))
-            .prefetch_related("type", "tenant", "terminations__site")
-        )
+            circuits_table = tables.CircuitTable(circuits)
+            circuits_table.columns.hide("provider")
 
-        circuits_table = tables.CircuitTable(circuits)
-        circuits_table.columns.hide("termination_a")
-        circuits_table.columns.hide("termination_z")
+            paginate = {
+                "paginator_class": EnhancedPaginator,
+                "per_page": get_paginate_count(request),
+            }
+            RequestConfig(request, paginate).configure(circuits_table)
 
-        paginate = {"paginator_class": EnhancedPaginator, "per_page": get_paginate_count(request)}
-        RequestConfig(request, paginate).configure(circuits_table)
+            return {
+                "circuits_table": circuits_table,
+            }
+        elif view_type == "list":
+            return {}
+        elif view_type == "bulk_edit":
+            return {}
+        else:
+            return {}
 
-        return {
-            "circuits_table": circuits_table,
-        }
 
+class ProviderViewSetRouter(NautobotRouter):
+    viewset = ProviderViewSet
 
-class ProviderNetworkEditView(generic.ObjectEditView):
-    queryset = ProviderNetwork.objects.all()
-    model_form = forms.ProviderNetworkForm
+    def __init__(self, prefix, basename):
+        super().__init__(prefix, basename)
```

Comment:
I'm seeing this overload over `__init__` in many places and I do not believe it's necessary, especially since you're not actually doing anything special there. It's completely sufficient to have a bare subclass with only the `viewset` attribute set.